### PR TITLE
chore(query): fix replace into from distributed select plan

### LINKED ISSUE #15655: bug: `replace into` not work as expected
**Summary**

`replace into test on(c) select sum(c) as c from source group by v` not work as expected in multi-nodes deployment.

failed with err msg: ResponseError with 1104: assertion failed: self.fragment_exchanges.is_empty()

query node version:  [v1.2.495-nightly](https://github.com/datafuselabs/databend/releases/tag/v1.2.495-nightly)

how to reproduce:

~~~
create or replace table test (c int);
create or replace table source (c int, v string);


insert into source select number as c, number::string from numbers(100000);

replace into test on(c) select sum(c) as c from source group by v; 
~~~

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

1. fix replace into from distributed select plan
2. add explain replace into
3. fix explain parser error

- Fixes #15655

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15657)

### CODE REVIEW

**Review by dantengsky — APPROVED** [review]

File: src/query/ast/src/ast/statements/replace.rs [hunk 1]
```diff
@@ -54,18 +54,18 @@ impl Display for ReplaceStmt {
...
             write_comma_separated_list(f, &self.columns)?;
-            write!(f, ") ")?;
+            write!(f, ")")?;
         }
-        write!(f, "ON CONFLICT")?;
+        write!(f, " ON CONFLICT")?;
         if !self.on_conflict_columns.is_empty() {
             write!(f, "(")?;
             write_comma_separated_list(f, &self.on_conflict_columns)?;
             write!(f, ") ")?;
```
Comment (TCeason) [inline src/query/ast/src/ast/statements/replace.rs:63]: write!(f, ") ")?;

modify to 

```
write!(f, " )")?;
```

File: src/query/ast/src/ast/statements/replace.rs [hunk 2]
```diff
@@ -54,18 +54,18 @@ impl Display for ReplaceStmt {
         if !self.columns.is_empty() {
             write!(f, "(")?;
             write_comma_separated_list(f, &self.columns)?;
-            write!(f, ") ")?;
+            write!(f, ")")?;
         }
-        write!(f, "ON CONFLICT")?;
+        write!(f, " ON CONFLICT")?;
         if !self.on_conflict_columns.is_empty() {
             write!(f, "(")?;
```
Comment (TCeason) [inline src/query/ast/src/ast/statements/replace.rs:61]: write!(f, " (")?;

File: src/query/ast/src/ast/statements/replace.rs (hunk 1 above)
Comment (sundy-li) [reply src/query/ast/src/ast/statements/replace.rs:63]: Let me fix this in next pr.

Because the current behavior is wrong, this is a fix.

**Review by TCeason — APPROVED** [review]

**Comment by BohuTANG** [comment]
```
Test finished, fail fast enabled, 1 out of 6865 records failed to run
0: statement failed: Databend sqllogictests error: http query error: code: 4000, Text: duplicated data detected in the values being replaced into (only the first one will be described): at row 2, ["id":1]
[SQL] replace into test on(id) select * from @test_stage (pattern => '.*parquet');
Error: SelfError("sqllogictest failed")
at tests/sqllogictests/suites/base/09_fuse_engine/09_0023_replace_into.test:299
```

File: src/query/service/src/interpreters/interpreter_replace.rs [hunk 3]
```diff
@@ -191,6 +192,12 @@ impl ReplaceInterpreter {
                 &mut purge_info,
             )
             .await?;
+
+        let is_value_source = matches!(self.plan.source, InsertInputSource::Values(_));
+        let replace_need_distributed = is_source_distributed
+            && self.ctx.get_settings().get_enable_distributed_replace()?
+            && !is_value_source;
```
Comment (xudong963) [inline src/query/service/src/interpreters/interpreter_replace.rs:199]: `is_source_distributed` contains `self.ctx.get_settings().get_enable_distributed_replace()?&& !is_value_source;`

File: src/query/service/src/interpreters/interpreter_replace.rs [hunk 4]
```diff
@@ -191,6 +192,12 @@ impl ReplaceInterpreter {
                 &mut purge_info,
             )
             .await?;
+
+        let is_value_source = matches!(self.plan.source, InsertInputSource::Values(_));
```
Comment (xudong963) [inline src/query/service/src/interpreters/interpreter_replace.rs:196]: `is_value_source` has been declared at line 177.

File: src/query/sql/src/planner/optimizer/optimizer.rs [hunk 5]
```diff
@@ -243,7 +243,9 @@ pub async fn optimize(opt_ctx: OptimizerContext, plan: Plan) -> Result<Plan> {
                     plan.source = InsertInputSource::SelectPlan(Box::new(optimized_plan));
                 }
                 InsertInputSource::Stage(p) => {
-                    let optimized_plan = optimize(opt_ctx.clone(), *p.clone()).await?;
+                    // seems we can't optimize this
+                    // let optimized_plan = optimize(opt_ctx.clone(), *p.clone()).await?;
```
Comment (xudong963) [inline src/query/sql/src/planner/optimizer/optimizer.rs:247]: Why?

**Review by xudong963 — APPROVED** [review]

**Pull request merged**